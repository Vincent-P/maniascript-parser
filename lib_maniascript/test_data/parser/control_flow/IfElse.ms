main()
{
    if (True) {
    }
    else if (True) {
    }
    else {
    }
}
