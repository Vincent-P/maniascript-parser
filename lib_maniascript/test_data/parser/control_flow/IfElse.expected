NODE_ROOT 0..76 {
  NODE_FUNC_DECL 0..76 {
    NODE_IDENTIFIER 0..4 {
      TOKEN_IDENT("main") 0..4
    }
    TOKEN_OPEN_PAREN("(") 4..5
    TOKEN_CLOSE_PAREN(")") 5..6
    TOKEN_WHITESPACE("\n") 6..7
    NODE_BLOCK 7..76 {
      TOKEN_OPEN_BRACE("{") 7..8
      TOKEN_WHITESPACE("\n    ") 8..13
      NODE_IF_ELSE 13..74 {
        TOKEN_IF("if") 13..15
        TOKEN_WHITESPACE(" ") 15..16
        TOKEN_OPEN_PAREN("(") 16..17
        NODE_EXPRESSION 17..21 {
          NODE_IDENTIFIER 17..21 {
            TOKEN_IDENT("True") 17..21
          }
        }
        TOKEN_CLOSE_PAREN(")") 21..22
        TOKEN_WHITESPACE(" ") 22..23
        NODE_BLOCK 23..30 {
          TOKEN_OPEN_BRACE("{") 23..24
          TOKEN_WHITESPACE("\n    ") 24..29
          TOKEN_CLOSE_BRACE("}") 29..30
        }
        TOKEN_WHITESPACE("\n    ") 30..35
        TOKEN_ELSE("else") 35..39
        TOKEN_WHITESPACE(" ") 39..40
        NODE_IF_ELSE 40..74 {
          TOKEN_IF("if") 40..42
          TOKEN_WHITESPACE(" ") 42..43
          TOKEN_OPEN_PAREN("(") 43..44
          NODE_EXPRESSION 44..48 {
            NODE_IDENTIFIER 44..48 {
              TOKEN_IDENT("True") 44..48
            }
          }
          TOKEN_CLOSE_PAREN(")") 48..49
          TOKEN_WHITESPACE(" ") 49..50
          NODE_BLOCK 50..57 {
            TOKEN_OPEN_BRACE("{") 50..51
            TOKEN_WHITESPACE("\n    ") 51..56
            TOKEN_CLOSE_BRACE("}") 56..57
          }
          TOKEN_WHITESPACE("\n    ") 57..62
          TOKEN_ELSE("else") 62..66
          TOKEN_WHITESPACE(" ") 66..67
          NODE_BLOCK 67..74 {
            TOKEN_OPEN_BRACE("{") 67..68
            TOKEN_WHITESPACE("\n    ") 68..73
            TOKEN_CLOSE_BRACE("}") 73..74
          }
        }
      }
      TOKEN_WHITESPACE("\n") 74..75
      TOKEN_CLOSE_BRACE("}") 75..76
    }
  }
}